specs/ledger/hs/cs-ledger.cabal

```diff
@@ -55,16 +55,27 @@ test-suite doctests
   default-language:    Haskell2010
   build-depends:       base
                      , doctest
-                     , doctest-discover
-                       --
+                       -- need these, when doctest-discover runs
+                       -- to ensure they are in scope.
+                     , bytestring
```

Comment:
@angerman the fact that we added these dependencies seem to indicate that they are only necessary for our `nix` infrastructure to successfully build. If that's the case, we might end up running into problems every time we run `weeder` to remove superfluous dependencies. 

Is there a way to avoid adding these?

Reply:
I don't know how `weeder` might respond to this :-/

I'm still confused why `doctest-discover` suggests putting doctest-discover into the dependencies. That makes absolutely no sense, as there is no code being imported but an executable is used.
 
https://github.com/input-output-hk/fm-ledger-rules/blob/25b3d3ac5c5bd3f76d77220c9765001db9a54523/specs/ledger/hs/test/DoctestDiscover.hs#L1

This even specifically calls out to a program at build time. So it would have to be 
```
build-tool-depends: doctest-discover:doctest-discover
```

As `doctest` is then essentially running a `ghci` session over the source it needs access to the packages.  I'll have to see why they are available to it in `cabal` by default.

That being said. I do agree that the addition of all the packages here is an annoyance and I'd rather not have us to do this. On the `build-tool-depends`, I believe the current behaviour of `cabal` is actually a bug, and I'd rather not rely on that. I know that @hvr has similar issues with `hspec-discover`. The implicit assumption seems to be that dependencies bring their executables into scope.